mars/services/web/ui/src/task_info/TaskTileableChart.js

```diff
@@ -0,0 +1,154 @@
+                this.g.setEdge(dependency.from_tileable_id, dependency.to_tileable_id, { label: "" });
+            });
+
+            let gInstance = this.g;
+            // Round the corners of the nodes
+            gInstance.nodes().forEach(function (v) {
+                var node = gInstance.node(v);
+                node.rx = node.ry = 5;
+            });
+
+            //makes the lines smooth
+            gInstance.edges().forEach(function (e) {
+                var edge = gInstance.edge(e.v, e.w);
+                edge.lineInterpolate = 'basis';
+
+                edge.style = "stroke: #333";
+            });
+
+            // Create the renderer
+            var render = new dagreD3.render();
+
+            // Set up an SVG group so that we can translate the final graph.
+            var svg = d3.select("#svg-canvas"),
+            inner = svg.append("g");
+
+            // Set up zoom support
+            var zoom = d3.zoom().on("zoom", function (e) {
+                inner.attr("transform", e.transform);
+            });
+            svg.call(zoom);
+
+            if (this.state.tileables.length !== 0) {
+                // Run the renderer. This is what draws the final graph.
+                render(inner, this.g);
+            }
+
+            const handleClick = (e, node) => {
+                const selectedTileable = this.state.tileables.filter((tileable) => tileable.tileable_id == node)[0];
+                this.setState({ selectedTileable: selectedTileable});
+            };
+
+            inner.selectAll("g.node").on("click", handleClick);
+
+            // Center the graph
+            var initialScale = 0.75;
+            svg.call(
+                zoom.transform,
+                d3.zoomIdentity.scale(initialScale)
+            );
+
+            if (this.g.graph() !== null || this.g.graph() !== undefined) {
+                svg.attr("height", this.g.graph().height * initialScale + 40);
+            } else {
+                svg.attr("height", 40);
+            }
+      }
+    };
+
+    render() {
+        return (
+            <Grid container spacing={3} >
+                 <Grid item xs={12}>
+                    <Title>Graph</Title>
+                </Grid>
+                <Grid item xs={12} md={6}>
+                    <Paper>
+                        <Grid item xs={12}>
+                            <svg
+                                id="svg-canvas"
+                                style={{ margin: 30, width: "100%", height: 700 }}
+                            />
+                        </Grid>
+                    </Paper>
+                </Grid>
+                <Grid item xs={12} md={6}>
+                    <Paper style={{ padding: 10 }}>
+                        <Grid item xs={12}>
+                            <Title>Tileable Detail: </Title>
+                        </Grid>
+                        <Grid item xs={12}>
+                            {
+                                this.state.selectedTileable
```

Comment:
It is better to create an independent `TileableDetail` class for further development.